### DISCUSSION TITLE
BPF filter issue

### DISCUSSION THREAD

**Discussion by mstannh** [Security-Onion-Solutions/securityonion#8970] [Unsupported Versions] [answered]
All,
I am receiving errors when applying the following BPF text to the .sls file at either the Global or minion level.
Here is a snippet:

>nids:
  bpf:
    - not dns 
   
In a separate attempt also tried with this filter

>-not host 192.169.12.1 

After saving the .sls file, I run the following command:

>  [root@seconmgr pillar]# salt 'seconsnsr*' state.highstate

I immediately receive the following error

>seconsnsr_sensor:
    Data failed to compile:
    The function "state.highstate" is running as PID 19972 and was started at 2022, Oct 21 14:49:16.226624 with jid 20221021144916226624
ERROR: Minions returned with non-zero exit code 

When I reconfigure the .sls file to remove  the "- not dns" line and run [root@seconmgr pillar]# salt 'seconsnsr*' state.highstate
It successfully completes as shown in the message below.

> Summary for seconsnsr_sensor
Succeeded: 236 (changed=20)
Failed:      0
Total states run:     236
Total run time:    60.815 s

Any ideas on what is wrong?
I am at 2.3.180.

**Comment by logz4u** [comment]
BPFs are not protocol aware. If you wanted to apply BPFs for DNS, you would need to specify the port or IP of the DNS server or something. So, you got the first message because your bpf input was not valid, so it errored out. However, your second input ( -not host 192.169.12.1 ) is valid, so you did not receive an error message. 

Anyways, if you want to exclude DNS, you could put

bpf:
  - not port 53
  - not IpAddressOfDNSServerOrSomething

Someone else should definitely feel free to correct me if I am wrong.

**Comment by mstannh** [comment]
Thanks for that clarification.
I reconfigured per your suggestion and tried the following.............which had no effect as both port 53 DNS and the IP are appearing on the Dashboard and in Kibana

> nids:
    bpf:
     - not port 53 ||
    - not host 192.168.12.1

**Reply by xfaith** [reply]
Check spacing, I believe the - should be two spaces away (could be just how you copied it though).  Also the II at the end I don't think should be there.

**Comment by mstannh** [comment, marked as answer]
Thank you all for the responses we appear to be filtering as intended 
It appears the use of the && not the || fixed my issue
